ui/components/src/components-react/tree/TreeDataProvider.ts

```diff
@@ -138,7 +138,7 @@ export interface EditableTreeDataProvider extends ITreeDataProvider {
 /**
  * MutableTreeDataProvider provides manipulation processing for the Tree.
  * Useful for Drag & Drop processing.
- * @beta
+ * @public
  */
 export interface MutableTreeDataProvider extends ITreeDataProvider {
```

Comment:
I'm hesitant on promoting `MutableTreeDataProvider` to `@public`. 

We removed the old `BeInspireTree`-based tree component and the `ControlledTree` component uses different means to handle mutations (`TreeModelSource.modifyModel`). In fact the `ITreeDataProvider.onTreeNodeChanged` event, which is a must for `MutableTreeDataProvider` implementations, is still marked `@deprecated`. So I'd rather move it to `@deprecated` than `@public`.

Adding @roluk and @DanEastBentley to hear their opinions.

Reply:
I agree that this should be @deprecated and a new standard interface or mechanism should be designed.

Reply:
This interface is only useful when used together with the deprecated Breadcrumbs component, thus it should transitively inherit `@deprecated` tag.

Reply:
I avoided the propertygrid and controlled subdirectories, leaving to @grigasp and his team to make the call there.

Reply:
@NancyMcCallB, do you want us to make the change in this PR or separate?